compiler/rustc_lint/src/pass_by_value.rs

```diff
@@ -0,0 +1,114 @@
+use crate::{LateContext, LateLintPass, LintContext};
+use rustc_errors::Applicability;
+use rustc_hir as hir;
+use rustc_hir::def::Res;
+use rustc_hir::def_id::DefId;
+use rustc_hir::{GenericArg, PathSegment, QPath, TyKind};
+use rustc_span::symbol::sym;
+
+declare_lint! {
+    /// The `pass_by_value` lint marks a type with `#[pass_by_value]` requiring it to always be passed by value.
+    /// This is usually used for types that are thin wrappers around references, so there is no benefit to an extra
+    /// layer of indirection. (Example: Ty which is a reference of TyCtx)
+    ///
+    /// ### Example
+    ///
+    /// ```rust
+    /// fn returns_result() -> Result<(), ()> {
+    ///     Ok(())
+    /// }
+    ///
+    /// fn main() {
+    ///     returns_result();
+    /// }
+    /// ```
```

Comment:
this example doesn't seem related to this lint :thinking:

Reply:
@lcnr ah sorry, it isn't, I haven't finished it up yet, I will write tests and update the example soon